packages/web/src/pages/chat-page/components/ChatMessageList.tsx

```diff
@@ -83,6 +99,22 @@ export const ChatMessageList = forwardRef<HTMLDivElement, ChatMessageListProps>(
     const chat = useSelector((state) => getChat(state, chatId))
     const userId = useSelector(accountSelectors.getUserId)
     const currentUserId = encodeHashId(userId)
+    const [unreadIndicatorEl, setUnreadIndicatorEl] =
+      useState<HTMLDivElement | null>(null)
+    const [, setLastScrolledChatId] = useState<string>()
+
+    const ref = useRef<HTMLDivElement>(null)
+
+    // A ref so that the unread separator doesn't disappear immediately when the chat is marked as read
```

Comment:
This seems like a behavior that will be implemented in mobile as well - is there any way we could encode the behavior around when to show the unread indicator in the slice layer so we don't have to re-implement tricks with refs on both platforms?

Reply:
Hmmm. I kinda like keeping this state localized to the component. I don't think this needs to bubble up to a global state. That could have real perf costs etc. and i think would be less intuitive.